test/testnet/unit/ODSafeManager.t.sol

```diff
@@ -0,0 +1,457 @@
+      vm.assume(_scenario.rando != addressesToAvoid[i]);
+    }
+  }
+}
+
+contract Unit_ODSafeManager_Deployment is Base {
+  function testDeployment() public {
+    assertEq(address(safeManager.vault721()), address(vault721), 'incorrect vault721');
+    assertEq(safeManager.safeEngine(), address(mockSafeEngine), 'incorrect safe engine');
+    assertEq(safeManager.taxCollector(), address(taxCollector), 'incorrect tax collector');
+  }
+}
+
+contract Unit_ODSafeManager_ViewFunctions is Base {
+  modifier happyPath(Scenario memory _scenario) {
+    _openSafe(_scenario);
+    _;
+  }
+
+  function test_getSafes_PerCollateral(Scenario memory _scenario) public happyPath(_scenario) {
+    uint256[] memory safes = safeManager.getSafes(_scenario.userProxy, _scenario.cType);
+
+    assertEq(safes.length, 1, 'incorrect number of safes');
+    assertEq(safes[0], 1, 'incorrect safe id');
+  }
+
+  function test_getSafes(Scenario memory _scenario) public happyPath(_scenario) {
+    uint256[] memory safes = safeManager.getSafes(_scenario.userProxy);
+
+    assertEq(safes.length, 1, 'incorrect number of safes');
+    assertEq(safes[0], 1, 'incorrect safe id');
+  }
+
+  function test_GetSafesData(Scenario memory _scenario) public happyPath(_scenario) {
+    (uint256[] memory _safes, address[] memory _safeHandlers, bytes32[] memory _cTypes) =
+      safeManager.getSafesData(_scenario.userProxy);
+
+    assertEq(_safes.length, 1, 'incorrect number of safes');
+    assertEq(_safeHandlers.length, 1, 'incorrect number of safe handlers');
+    assertEq(_cTypes.length, 1, 'incorrect number of cTypes');
+  }
+}
+
+contract Unit_ODSafeManager_SAFEManagement is Base {
+  event OpenSAFE(address indexed _sender, address indexed _own, uint256 indexed _safe);
+
+  modifier happyPath(Scenario memory _scenario) {
+    _assumeHappyPath(_scenario);
+    _openSafe(_scenario);
+    _;
+  }
+
+  function test_OpenSafe(Scenario memory _scenario) public {
+    vm.startPrank(_scenario.user);
+
+    _scenario.userProxy = vault721.build(_scenario.user);
+
+    vm.expectEmit(true, true, false, true);
+
+    emit OpenSAFE(_scenario.user, _scenario.userProxy, 1);
+
+    vm.mockCall(address(vault721), abi.encodeWithSelector(IVault721.mint.selector), abi.encode());
+
+    _scenario.safeId = safeManager.openSAFE('i', _scenario.userProxy);
+
+    assertEq(_scenario.safeId, 1, 'incorrect safeId returned');
+  }
+
+  event AllowSAFE(address indexed _sender, uint256 indexed _safe, address _usr, bool _ok);
+
+  function test_AllowSafe(Scenario memory _scenario) public happyPath(_scenario) {
+    vm.startPrank(_scenario.userProxy);
+
+    vm.expectEmit();
+    emit AllowSAFE(_scenario.userProxy, _scenario.safeId, owner, true);
+
+    safeManager.allowSAFE(_scenario.safeId, owner, true);
+    uint256[] memory _safes = safeManager.getSafes(_scenario.userProxy);
+    assertEq(_safes.length, 1, 'incorrect number of safes');
+    assertEq(_safes[0], 1, 'incorrect safe id');
+  }
+
```

Comment:
Should check the state of safeCan here instead of getSafes

Reply:
whoops. fixed.